app/models/name/merge.rb

```diff
@@ -130,4 +134,40 @@ def merge(old_name)
     # Finally destroy the name.
     old_name.destroy
   end
+
+  ##############################################################################
+
+  private
+
+  def approved_synonym_of_proposed_name?
+    !deprecated && (other_synonym_ids & Naming.all.pluck(:name_id)).present?
+  end
+
+  def correctly_spelled_ancestor_of_proposed_name?
+    return false if correct_spelling.present?
+    return above_genus_is_ancestor? if classified_above_genus?
+    return genus_or_species_is_ancestor? if [:Genus, :Species].include?(rank)
+    return group_is_ancestor? if rank == :Group
+
+    false
+  end
+
+  def above_genus_is_ancestor?
+    Name.joins(:namings).where(
+      "classification LIKE ?", "%#{rank}: #{classification_name}%"
+    ).any?
+  end
+
+  def genus_or_species_is_ancestor?
+    Name.joins(:namings).where("text_name LIKE ?", "#{text_name}%").
+      ranked_below(rank).any?
+  end
+
+  # Better than nothing for preventing accidental deletion of a Group.
+  # although it catches only a few cases
```

Comment:
I'm not sure I would actually even try to prevent this case.  Do you have a use case in mind?  My go-to use case for groups is always Lecidea atrobrunnea group, an artificial group with species in multiple unrelated genera.  All of those names, if proposed, would continue to be just fine even if I deleted "Lecidea atrobrunnea group".  No name should ever directly include a pseudotaxon like a group name in their classification string.  I don't see how it would break anything.

Reply:
I'll get rid of it. It only catches a few cases and complicates things.
FYI, 
- The typical use cases I see are:
  - species complexes (multiple cryptic species)
  - group of non-cryptic species that are hard/impossible to ID without microscopy and/or molecular work. 
-  Sometimes, users go to extra effort to describe the group, provide taxonomic info, etc. It would be too bad to loose that. But on balance, we're better off without the complexity
- I don't think it's possible to include a Group in a classification string.